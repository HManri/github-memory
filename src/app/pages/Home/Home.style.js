import styled from 'styled-components';

export const HomeWrapper = styled.div`
  position: relative;
  height: 100%;
  background-color: ${(props) => props.theme.colors.wine};
`;

export const HomeFloating = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: center;
  height: 95%;
  width: 95%;
  background-color: ${(props) => props.theme.colors.greengrey};
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: ${(props) => props.theme.spacings.big}px;
  border-radius: 16px;
`;

export const Title = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: ${(props) => props.theme.spacings.small}px;
`;

export const MainTitle = styled.div`
  ${(props) => props.theme.text.title};
  color: ${(props) => props.theme.colors.lightBlack};
`;

export const StrongTitle = styled.div`
  ${(props) => props.theme.text.title};
  font-weight: 600;
`;

export const ButtonText = styled.span`
  ${(props) => props.theme.text.h1};
  text-transform: uppercase;
`;

export const HomeActions = styled.div``;
