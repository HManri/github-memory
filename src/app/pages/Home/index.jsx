import React from 'react';
import { useNavigate } from 'react-router-dom';

import Button from 'components/Button';
import {
  HomeWrapper,
  HomeFloating,
  Title,
  MainTitle,
  StrongTitle,
  HomeActions,
  ButtonText,
} from './Home.style';

export default function Home() {
  const navigate = useNavigate();

  return (
    <HomeWrapper>
      <HomeFloating>
        <Title>
          <MainTitle>Welcome to</MainTitle>
          <StrongTitle>GitHub Memory</StrongTitle>
        </Title>

        <HomeActions>
          <Button testId="start_game__btn" onClick={() => navigate('/match')}>
            <ButtonText>Start</ButtonText>
          </Button>
        </HomeActions>
      </HomeFloating>
    </HomeWrapper>
  );
}
