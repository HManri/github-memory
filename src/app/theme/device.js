const size = {
  mobile: '425px',
  desktop: '768px',
};

export const device = {
  mobile: `(min-width: ${size.mobile})`,
  desktop: `(min-width: ${size.desktop})`,
};
