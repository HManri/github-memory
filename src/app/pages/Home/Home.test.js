import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from 'theme/DefaultTheme';
import Home from '.';

describe('Home', () => {
  test('should render correctly', () => {
    const home = render(
      <BrowserRouter>
        <ThemeProvider theme={theme}>
          <Home testId="home-test" />
        </ThemeProvider>
      </BrowserRouter>,
    );

    expect(home).toMatchSnapshot();
  });
});
