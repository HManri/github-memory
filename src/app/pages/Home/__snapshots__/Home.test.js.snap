// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gswNZR bAwMWr"
        data-testid="home"
      >
        <div
          class="sc-dkrFOg ilenFF"
          data-testid="home__box"
        >
          <div
            class="sc-hLBbgP iuiwzT"
            data-testid="home__title"
          >
            <div
              class="sc-eDvSVe llhKVC"
            >
              Welcome to
            </div>
            <div
              class="sc-jSUZER AzuVu"
            >
              GitHub Memory
            </div>
          </div>
          <div
            class="sc-iBYQkv"
            data-testid="home__actions"
          >
            <button
              class="sc-bcXHqe kJpEhf"
              data-testid="start_game__btn"
            >
              <span
                class="sc-gKPRtg hPtRSE"
              >
                Start
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gswNZR bAwMWr"
      data-testid="home"
    >
      <div
        class="sc-dkrFOg ilenFF"
        data-testid="home__box"
      >
        <div
          class="sc-hLBbgP iuiwzT"
          data-testid="home__title"
        >
          <div
            class="sc-eDvSVe llhKVC"
          >
            Welcome to
          </div>
          <div
            class="sc-jSUZER AzuVu"
          >
            GitHub Memory
          </div>
        </div>
        <div
          class="sc-iBYQkv"
          data-testid="home__actions"
        >
          <button
            class="sc-bcXHqe kJpEhf"
            data-testid="start_game__btn"
          >
            <span
              class="sc-gKPRtg hPtRSE"
            >
              Start
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
